### DISCUSSION TITLE
are there any recording small "segment_time" problems?

### DISCUSSION THREAD

**Discussion by spacebares** [blakeblackshear/frigate#4405] [Ask A Question] [answered]
Aside from maybe I/O having to manage a larger amount of files for read & writing, is there any other penalty to having a small `segment_time` for frigate record? I haven't noticed any increased cpu usage, and if there is its so negligible.

The default is 10 second `segment_time` i believe? For event viewing if you view the event too soon, sometimes you get an error saying the clip is unavailable or couldn't generate clip for camera-Name etc. Especially for home assistant integration API.

The `segment_time` here is what i mean:
```
ffmpeg:
  output_args:
    record: -f segment -segment_time 1 -segment_format mp4 -reset_timestamps 1 -strftime 1 -c:v copy -c:a aac
```

I set it to 1 second and now i am able to view clips for events that are "in progress" where previously it would error out saying no clip available. This has provided great reliability for mobile notifications as you can view the mp4 in progress with no worry of an error. 
Even frigate_hass_card's recent clip mode when viewed in that moment, starts playing the event that is in progress.

Of course if you view the event is in progress while its say 5 seconds in, you are only presented with 5 seconds of video then the clip ends. Even though the event is still going on. You have to manually refresh the page to keep getting fed more of the video file. **Would be nice to have some kind of way to "stream" the event clip if the event is in the "in-progress" state.**

**Comment by NickM-27** [comment, marked as answer]
> Would be nice to have some kind of way to "stream" the event clip if the event is in the "in-progress" state.

This is definitely supported, if you're trying to view the event as soon as it started you of course have to wait until that (by default) 10 second segment is out of the cache, but it is 100% possible to play back in progress clips. 

The downside of the 1-second segment sizes are going to be retention. Now instead of 10 seconds being stored if any of those 10 seconds have motion (or however your retention is setup), every 1 second segment has to have motion or active objects (again depending on config) for it to be kept so you could end up with a lot of skipping 1 second here or there. It's also more work to build the streaming m3u8 as it has to account for a lot more segments. You're also going to see issues depending on your camera as a segment can only be created where it starts on a key frame so some cameras that have inconsistent key frames will cause issues there too.

**Reply by spacebares** [reply]
Confirmed, 1 second clips has lots of skipping in some events. Increased to 3 seconds has eliminated all skipping. I suppose its based on system & camera at some level.
